src/shaders/line.fragment.glsl

```diff
@@ -58,11 +58,12 @@ void main() {
     alpha *= smoothstep(0.5 - sdfgamma / floorwidth, 0.5 + sdfgamma / floorwidth, sdfdist);
 #endif
 
+    highp vec4 out_color;
```

Comment:
Now these should be detected during compile time in Metal. It might still need follow-up PRs in GL-JS if we don't test all the new GL-JS shader changes on the side of GL-Native, but we'll test those at least when we bump the repo. When I did https://github.com/mapbox/mapbox-gl-js/pull/12249 it was still only supported for a small subset of shaders, now all of them are using this approach.

Later on we could also try to use a different GLSL version which supports specialization constants, so then we could write the conditional blocks using those. But then we'll need to use a cross-compiler in GL-JS to get a version 100 output of those, where we would replace those blocks with ifdefs.